src/components/EventCreationModal.tsx

```diff
@@ -0,0 +1,101 @@
+import { useState } from "react";
+
+export function EventCreationModal(props: {
+	showEventCreationModal: boolean;
+	handleCreateEventClick: Function;
```

Comment:
Since you're using TypeScript, it would be nice to properly type these functions. Saying `Function` is basically saying like `any` for functions.
The type here should probably be `() => void`